kv/txn_coord_sender_test.go

```diff
@@ -114,20 +78,18 @@ func TestTxnCoordSenderAddRequest(t *testing.T) {
 	defer s.Stop()
 	defer teardownHeartbeats(s.Sender)
 
-	txn := newTxn(s.Clock, roachpb.Key("a"))
-	put, h := createPutRequest(roachpb.Key("a"), []byte("value"), txn)
+	txn := client.NewTxn(*s.DB)
 
 	// Put request will create a new transaction.
-	reply, err := client.SendWrappedWith(s.Sender, nil, h, put)
-	if err != nil {
+	if err := txn.Put(roachpb.Key("a"), []byte("value")); err != nil {
 		t.Fatal(err)
 	}
-	txnMeta, ok := s.Sender.txns[string(txn.ID)]
+	txnMeta, ok := s.Sender.txns[string(txn.Proto.ID)]
 	if !ok {
 		t.Fatal("expected a transaction to be created on coordinator")
 	}
-	if !reply.Header().Txn.Writing {
-		t.Fatal("response Txn is not marked as writing")
+	if !txn.Proto.Writing {
```

Comment:
ah, I stand corrected. This is doing exactly the right thing, albeit a little less directly. Carry on.